ipamd/datastore/data_store.go

```diff
@@ -285,37 +285,36 @@ func (ds *DataStore) assignPodIPv4AddressUnsafe(k8sPod *k8sapi.K8SPodInfo) (stri
 			if k8sPod.IP == addr.address {
 				// After L-IPAM restart and built IP warm-pool, it needs to take the existing running pod IP out of the pool.
 				if !addr.Assigned {
-					ds.assigned++
-					eni.AssignedIPv4Addresses++
-					assignedIPs.Set(float64(ds.assigned))
-					addr.Assigned = true
+					increaseAssignedCount(ds, eni, addr)
 				}
-				ds.podsIP[podKey] = PodIPInfo{IP: addr.address, DeviceNumber: eni.DeviceNumber}
 				log.Infof("AssignPodIPv4Address Reassign IP %v to pod (name %s, namespace %s)",
 					addr.address, k8sPod.Name, k8sPod.Namespace)
+				ds.podsIP[podKey] = PodIPInfo{IP: addr.address, DeviceNumber: eni.DeviceNumber}
```

Comment:
Did you swap the order since the above says reassign vs reassigned? Also may be good to add the colon after AssignPodIPv4Address